### PR TITLE
fix(js): show the new messages pill when there are more than x notifications

### PR DESCRIPTION
**Pull request by LetItRock**
### What changed? Why was the change needed?

Inbox: improve the new notifications CTA

### Screenshots

**The cases:**

1. The notifications were loaded but the Inbox was later closed. 
 
https://github.com/user-attachments/assets/f66aea7b-470b-454a-a5d9-e4d742fe12e7

2. Events were sent while the Inbox was closed (never opened).
 
https://github.com/user-attachments/assets/394b9a03-7bc9-40b6-acbb-a86c1c5317b5

3. Events were sent while the Inbox was opened.

https://github.com/user-attachments/assets/3c5f7ab2-6c79-4357-b767-d023d10b4f74

4. Tabs: events were sent while the Inbox was closed (never opened).

https://github.com/user-attachments/assets/3190a2a8-fa0f-41dc-8a94-79e7ec70db7c

5. Tabs: events were sent while the Inbox was opened.

https://github.com/user-attachments/assets/cf65f1e9-5a55-47c7-9817-219182d4bb90

### CODE REVIEW

File: packages/js/src/ui/components/Inbox.tsx [hunk 1]
```diff
@@ -89,9 +89,11 @@ export const InboxContent = (props: InboxContentProps) => {
 
 export const Inbox = (props: InboxProps) => {
   const style = useStyle();
+  const { isOpened, setIsOpened } = useInboxContext();
```
Comment (LetItRock) [inline packages/js/src/ui/components/Inbox.tsx:92]: Storing the `isOpened` popover flag in the InboxContext

File: packages/js/src/ui/components/Inbox.tsx [hunk 2]
```diff
@@ -89,9 +89,11 @@ export const InboxContent = (props: InboxContentProps) => {
...
+  const isOpen = () => props?.open ?? isOpened();
```
Comment (LetItRock) [inline packages/js/src/ui/components/Inbox.tsx:93]: Use the controlled prop or uncontrolled state

File: packages/js/src/ui/components/primitives/Popover/PopoverRoot.tsx [hunk 3]
```diff
@@ -24,6 +24,9 @@ type PopoverContextValue = {
 const PopoverContext = createContext<PopoverContextValue | undefined>(undefined);
 
 export function PopoverRoot(props: PopoverRootProps) {
+  const [uncontrolledIsOpen, setUncontrolledIsOpen] = createSignal(props.open ?? false);
+  const onOpenChange = () => props.onOpenChange ?? setUncontrolledIsOpen;
+  const open = () => props.open ?? uncontrolledIsOpen();
```
Comment (LetItRock) [inline packages/js/src/ui/components/primitives/Popover/PopoverRoot.tsx:29]: Fixes to the PopoverRoot to operate on update prop `open`.

File: packages/js/src/ui/context/CountContext.tsx [hunk 4]
```diff
@@ -68,9 +70,14 @@ export const CountProvider = (props: ParentProps) => {
     const notificationsCache = novu.notifications.cache;
     const limitValue = limit();
     const tabFilter = { ...filter(), tags, offset: 0, limit: limitValue };
+    const hasEmptyCache = !notificationsCache.has(tabFilter);
+    if (!isOpened() && hasEmptyCache) {
+      return;
+    }
```
Comment (LetItRock) [inline packages/js/src/ui/context/CountContext.tsx:76]: If the popover is not opened and has empty cache then do noting (do not update cache, either counts)

File: packages/js/src/ui/context/CountContext.tsx [hunk 5]
```diff
@@ -68,9 +70,14 @@ export const CountProvider = (props: ParentProps) => {
...
+
     const cachedData = notificationsCache.getAll(tabFilter) || { hasMore: false, filter: tabFilter, notifications: [] };
-    const hasLessThenTenItems = (cachedData?.notifications.length || 0) < limitValue;
-    if (hasLessThenTenItems) {
+    const hasLessThenMinAmount = (cachedData?.notifications.length || 0) < MIN_AMOUNT_OF_NOTIFICATIONS;
+    if (hasLessThenMinAmount) {
```
Comment (LetItRock) [inline packages/js/src/ui/context/CountContext.tsx:80]: When we don't have anything in the cache or it's less then one item, put the new notification to the cache (show in the view).

**Review by BiswaViraj — APPROVED** [review]

**Review by desiprisg — APPROVED** [review]

**Review by SokratisVidros — APPROVED** [review]

**Pull request merged**